### PR TITLE
copr: vectorize PeriodAdd, PeriodDiff

### PR DESCRIPTION
**Pull request by MaiCw4J**
Signed-off-by: mak <610546069@qq.com>

PCP #5751 
###  What have you changed?

Add vectorized PeriodAdd, PeriodDiff

###  What is the type of the changes?

Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No. It should not change the behavior.

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/88

**Comment by breeswish** [comment]
@iosmanthus @zhangjinpeng1987 PTAL, thanks!

File: components/tidb_query/src/expr_util/time.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    let (year, month) = (period / 100, period % 100);
+    if year < 70 {
+        (year + 2000) * 12 + month - 1
+    } else if year < 100 {
+        (year + 1900) * 12 + month - 1
+    } else {
+        year * 12 + month - 1
+    }
+}
+
```
Comment (iosmanthus) [inline components/tidb_query/src/expr_util/time.rs]: what if the `period` is pretty large, like: `1234567891011`.
in mysql, it reports:
```sql
+------------------------------+
| period_add(1234567891011, 0) |
+------------------------------+
| 0                            |
+------------------------------+

```

**Comment by iosmanthus** [comment]
@MaiCw4J please handle the conflicts~

File: components/tidb_query/src/expr_util/time.rs (hunk 1 above)
Comment (MaiCw4J) [reply components/tidb_query/src/expr_util/time.rs]: What version of mysql did you get? Tested both in version **5.7.22** and **5.6.20-log** , i get the same result as the push-down functions

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/88

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/88

**Comment by breeswish** [comment]
Please fix conflicts. Thanks!

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/88

File: components/tidb_query/src/expr_util/time.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! Common utility implementations for time expression
+
+pub fn period_to_month(period: u64) -> u64 {
```
[the remark this answers is not in the document]
Comment (iosmanthus) [reply components/tidb_query/src/expr_util/time.rs]: @MaiCw4J Please move these function to `DateTime` instead.

File: components/tidb_query/src/expr_util/time.rs (hunk 1 above)
Comment (iosmanthus) [reply components/tidb_query/src/expr_util/time.rs]: `mysql  Ver 15.1 Distrib 10.4.10-MariaDB, for Linux (x86_64) using readline 5.1`

**Comment by breeswish** [comment]
Conflicts need to be fixed again :(

**Comment by MaiCw4J** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/88

**Pull request merged**